server/src/infra/repositories/media.repository.ts

```diff
@@ -23,11 +25,28 @@ export class MediaRepository implements IMediaRepository {
   }
 
   async resize(input: string | Buffer, output: string, options: ResizeOptions): Promise<void> {
-    await sharp(input, { failOn: 'none' })
-      .resize(options.size, options.size, { fit: 'outside', withoutEnlargement: true })
-      .rotate()
-      .toFormat(options.format)
-      .toFile(output);
+    if (typeof input === 'string' && unsupportedSharpTypes.has(mimeTypes.lookup(input))) {
```

Comment:
Can you refactor this into another helper method in the `domain.constants.ts` and include it into the mimeTypes on line 101 of that file?